src/Nethermind/Nethermind.State/Snap/PathWithStorageSlot.cs

```diff
@@ -1,19 +1,31 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
+using System.Numerics;
 using Nethermind.Core.Crypto;
 
 namespace Nethermind.State.Snap
 {
-    public class PathWithStorageSlot
+    public readonly struct PathWithStorageSlot(ValueHash256 keyHash, byte[] slotRlpValue)
+        : IEquatable<PathWithStorageSlot>, IEqualityOperators<PathWithStorageSlot, PathWithStorageSlot, bool>
     {
-        public PathWithStorageSlot(ValueHash256 keyHash, byte[] slotRlpValue)
+        public ValueHash256 Path { get; } = keyHash;
+        public byte[] SlotRlpValue { get; } = slotRlpValue;
+
+        public bool Equals(in PathWithStorageSlot other)
         {
-            Path = keyHash;
-            SlotRlpValue = slotRlpValue;
+            return Path == other.Path && SlotRlpValue.AsSpan().SequenceEqual(other.SlotRlpValue);
         }
 
-        public ValueHash256 Path { get; set; }
-        public byte[] SlotRlpValue { get; set; }
+        public bool Equals(PathWithStorageSlot other) => Equals(in other);
```

Comment:
Interesting, so the first equal only is not enough?

Reply:
Need to implement `Equals(object)` for the tests or fails with default; so added `in` version since multiple equals forwarding to the same implementation since is chunky struct